src/ray/gcs/gcs_server/gcs_placement_group_manager.cc

```diff
@@ -264,6 +270,13 @@ void GcsPlacementGroupManager::RegisterPlacementGroup(
                                        placement_group);
   AddToPendingQueue(placement_group);
 
+  if (placement_group->IsDetached()) {
```

Comment:
User `creation` and `destroy` listeners to instead of the directly dependency of `gcs_virtual_cluster_manager_`